cabal-ghc921.project

```diff
@@ -78,191 +61,3 @@ allow-newer:
   diagrams:diagrams-core,
   Chart-diagrams:diagrams-core,
   SVGFonts:diagrams-core
-
-constraints:
-  Agda ==2.6.1.3,
-  Diff ==0.4.0,
-  EdisonAPI ==1.3.1,
-  EdisonCore ==1.3.2.1,
-  FPretty ==1.1,
-  HTTP ==4000.3.16,
-  HUnit ==1.6.2.0,
-  QuickCheck ==2.14.2,
```

Comment:
did you intend to delete all the constraints?

Reply:
Yep, afaiu we dont need them having pinned the hackage index. They usually live in a cabal.project.freeze and no in the main one,. But maybe @fendor could expand the rationale behind

Reply:
IIRC, these are constraints from head.hackage to make sure the patched versions are used. I *think* removal of them is fine.

Reply:
Oh, i see, maybe it would be nice to add another cabal-9.2.1.project.local with all the info needed to use head.hackage, the head.hackage repository info itself (it is advised in the head.hackage repo to set it per project/cabal.project.local) + constraints
So any developer could use such cabal-9.2.1.project.local out of the box

Reply:
If you prefer an additional file containing it, fine by me. BTW, these constraints might be outdated, unsure how to best proceed with updating these.

Reply:
`cabal freeze` having head.hackage as a overlay repo could help? maybe we could add the head.hackage repo in cabal-921.project if it is needed to make it work for sure (and remove it when patches are upstreamed)

And use directly the constraints it offer: https://ghc.gitlab.haskell.org/head.hackage/cabal.constraints 
Or use `head.hackage.sh init-local`

We also could check if `index-state` affects head.hackage so it could help to avoid the need of constraints